core/src/main/java/org/frankframework/pipes/Json2XmlValidator.java

```diff
@@ -186,60 +186,125 @@ private static double getQuality(String mimeType) {
 	 */
 	@Override
 	public PipeRunResult doPipe(Message input, PipeLineSession session, boolean responseMode, String messageRoot) throws PipeRunException {
-		String messageToValidate;
+		DocumentFormat inputFormat;
+		Character firstChar;
 		try {
-			messageToValidate = Message.isNull(input) ? "{}" : input.asString();
-		} catch (IOException e) {
-			throw new PipeRunException(this, "cannot open stream", e);
+			firstChar = findFirstCharacter(input);
```

Comment:
is het niet mooier om deze in de determineDocumentFormat methode te hebben, dan geef je maar 1 arg mee :)

Reply:
Ja, veel mooier. Daarom deed ik het eerst ook daar.
Maar bij het maken van de error-message heb je hem weer opnieuw nodig dus ik wilde hem niet weer opnieuw ophalen in dat geval.
Daarom heb ik hem nu toch maar hier als parameter toegevoegd. Scheelt ook weer een IOException die je moet afvangen op een plek waar die toch eigenlijk niet meer kan optreden.